### PR TITLE
Modify check on resourceGroup.location

### PR DESCRIPTION
**Pull request by zjpjack**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: test/solution-template-validation-tests/test/mainTemplateTests.js [hunk 1]
```diff
@@ -79,15 +79,16 @@ describe('template files - ', () => {
 
         });
 
-        /** The location parameter MUST NOT have allowedValues property and MUST have defaultValue property whose value MUST be '[resourceGroup().location]' */
+        /** The location parameter MUST NOT have allowedValues property and if it has defaultValue property whose value MUST be '[resourceGroup().location]' */
```
Comment (vivsriaus) [inline test/solution-template-validation-tests/test/mainTemplateTests.js]: The location parameter MUST NOT have allowedValues property and if it's value MUST be '[resourceGroup().location]'

File: test/solution-template-validation-tests/test/mainTemplateTests.js [hunk 2]
```diff
@@ -118,6 +119,7 @@ describe('template files - ', () => {
             /** resourceGroup().location should NOT be present anywhere in template, EXCEPT as a defaultValue */
             it.each(templateObject, 'resourceGroup().location must NOT be be used in the template file ' + templateJSONObject.filename + ', except as a default value for the location parameter. Please correct similar errors in the file', function(element, next) {
                 var templateFileContent = JSON.stringify(templateObject);
```
Comment (vivsriaus) [inline test/solution-template-validation-tests/test/mainTemplateTests.js]: Please check if there is a "minify" json method that removes spaces/carriage return, so we wouldn't have to check for space/special chars in our regexes

Comment (zjpjack) [reply test/solution-template-validation-tests/test/mainTemplateTests.js]: Unfortunately, the stringify is the only json method to transfer json to string. If we would like a method that removes spaces/carriage return, we need to implement one in util.js.

File: test/solution-template-validation-tests/test/mainTemplateTests.js (hunk 1 above)
Comment (zjpjack) [reply test/solution-template-validation-tests/test/mainTemplateTests.js]: Here we need to check two properties, "allowedValues" and "defaultValue", so I think the original comment is more accurate.

Comment (vivsriaus) [reply test/solution-template-validation-tests/test/mainTemplateTests.js]: /** The location parameter MUST NOT have allowedValues property and if it has defaultValue property, it's value MUST be '[resourceGroup().location]' */

**Pull request merged**